lib/eclipse/EclipseState/Aquifer.cpp

```diff
@@ -0,0 +1,338 @@
+        }
+
+        return res;
+    }
+
+    void Completion::fixDefaultIJ(int wellHeadI , int wellHeadJ) {
+        if (m_i < 0)
+            m_i = wellHeadI;
+
+        if (m_j < 0)
+            m_j = wellHeadJ;
+    }
+
+    void Completion::shift_complnum( int shift ) {
+        this->m_complnum += shift;
+    }
+
+    int Completion::getI() const {
+        return m_i;
+    }
+
+    int Completion::getJ() const {
+        return m_j;
+    }
+
+    int Completion::getK() const {
+        return m_k;
+    }
+
+    int Completion::complnum() const {
+        return this->m_complnum;
+    }
+
+    WellCompletion::StateEnum Completion::getState() const {
+        return m_state;
+    }
+
+    double Completion::getConnectionTransmissibilityFactor() const {
+        return m_connectionTransmissibilityFactor.getValue();
+    }
+
+    double Completion::getDiameter() const {
+        return m_diameter.getValue();
+    }
+
+    double Completion::getSkinFactor() const {
+        return m_skinFactor.getValue();
+    }
+
+    int Completion::getSatTableId() const {
+        return m_satTableId;
+    }
+
+    const Value<double>& Completion::getConnectionTransmissibilityFactorAsValueObject() const {
+        return m_connectionTransmissibilityFactor;
+    }
+
+    Value<double> Completion::getDiameterAsValueObject() const {
+        return m_diameter;
+    }
+
+    Value<double> Completion::getSkinFactorAsValueObject() const {
+        return m_skinFactor;
+    }
+
+    WellCompletion::DirectionEnum Completion::getDirection() const {
+        return m_direction;
+    }
+
+    double Completion::getWellPi() const {
+        return m_wellPi;
+    }
+
+    int Completion::getSegmentNumber() const {
+        if (!attachedToSegment()) {
+            throw std::runtime_error(" the completion is not attached to a segment!\n ");
+        }
+        return m_segment_number;
+    }
+
+    double Completion::getCenterDepth() const {
+        return m_center_depth;
+    }
+
+    bool Completion::attachedToSegment() const {
+        return (m_segment_number > 0);
+    }
+
+    bool Completion::operator==( const Completion& rhs ) const {
+        return this->m_i == rhs.m_i
+            && this->m_j == rhs.m_j
+            && this->m_k == rhs.m_k
+            && this->m_complnum == rhs.m_complnum
+            && this->m_diameter == rhs.m_diameter
+            && this->m_connectionTransmissibilityFactor
+               == rhs.m_connectionTransmissibilityFactor
+            && this->m_wellPi == rhs.m_wellPi
+            && this->m_skinFactor == rhs.m_skinFactor
+            && this->m_satTableId == rhs.m_satTableId
+            && this->m_state == rhs.m_state
+            && this->m_direction == rhs.m_direction
+            && this->m_segment_number == rhs.m_segment_number
+            && this->m_center_depth == rhs.m_center_depth;
+    }
+
+    bool Completion::operator!=( const Completion& rhs ) const {
+        return !( *this == rhs );
+    }
+}
```

Comment:
This looked very weird - the existing `Completion` in a new file called `Aquiffer.c`?